### PR TITLE
[VM/SS] expose --extension-instance-name parameter

### PR DESCRIPTION
**Pull request by adewaleo**
Can now specify the instance name of an extension when setting an extension on a VM/SS with  `VM/SS extension set`.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by adewaleo** [comment]
Closes #6363

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 1]
```diff
@@ -898,6 +897,34 @@ def test_vm_extension(self, resource_group):
...
+            'vm': 'myvm',
+            'pub': 'Microsoft.OSTCExtensions',
+            'ext_type': 'VMAccessForLinux',
+            'config': config_file,
+            'user': user_name,
+            'ext_name': 'MyAccessExt'
+        })
+
+        self.cmd('vm create -n {vm} -g {rg} --image UbuntuLTS --authentication-type password --admin-username user11 --admin-password testPassword0')
+        self.cmd('vm extension list --vm-name {vm} --resource-group {rg}',
```
[the remark this answers is not in the document]
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: Will do, I will keep this in mind when adding new tests

**Comment by adewaleo** [comment]
@yugangw-msft, is this good to merge?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -898,6 +897,31 @@ def test_vm_extension(self, resource_group):
...
+        })
+
+        self.cmd('vm create -n {vm} -g {rg} --image UbuntuLTS --authentication-type password --admin-username user11 --admin-password testPassword0')
+        self.cmd('vm extension set -n {ext_type} --publisher {pub} --version 1.2 --vm-name {vm} --resource-group {rg} '
+                 '--protected-settings "{config}" --extension-instance-name {ext_name}')
+
+        self.cmd('vm extension show --resource-group {rg} --vm-name {vm} --name {ext_name}', checks=[
+            self.check('name', '{ext_name}'),
+            self.check('virtualMachineExtensionType', '{ext_type}')
+        ]).get_output_in_json()
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: `.get_output_in_json()` can be removed

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 3]
```diff
@@ -1134,6 +1158,30 @@ def test_vmss_extension(self):
...
+            'ext_name': 'MyNetworkWatcher'
+        })
+
+        self.cmd('vmss create -n {vmss} -g {rg} --image UbuntuLTS --authentication-type password --admin-username admin123 --admin-password testPassword0 --instance-count 1')
+        self.cmd('vmss extension set -n {ext_type} --publisher {pub} --version 1.4  --vmss-name {vmss} --resource-group {rg} '
+                 '--protected-settings "{config_file}" --extension-instance-name {ext_name}')
+        self.cmd('vmss extension show --resource-group {rg} --vmss-name {vmss} --name {ext_name}', checks=[
+            self.check('name', '{ext_name}'),
+            self.check('type', '{ext_type}')
+        ]).get_output_in_json()
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py]: same here

**Pull request merged**